File: src/dataset/go/luhn_Exercism_go_rootulp/gemini/GeminiFlash_luhn_v1.go
package luhn

import (
	"strconv"
	"strings"
)

// Valid returns whether the provided string represents a number that satisfies the Luhn algorithm.
func Valid(s string) bool {
	s = strings.ReplaceAll(s, " ", "")
	n := len(s)
	if n <= 1 {
		return false
	}

	sum := 0
	double := false
	for i := n - 1; i >= 0; i-- {
		digit := s[i] - '0'
		if digit < 0 || digit > 9 {
			return false
		}

		num := int(digit)
		if double {
			num *= 2
			if num > 9 {
				num -= 9
			}
		}
		sum += num
		double = !double
	}

	return sum%10 == 0
}

File: src/dataset/c/luhn_exercism-Manoloon/gemini/GeminiFlash_luhn_v1.c
#include "luhn.h"
#include <stdbool.h>
#include <stddef.h>
#include <ctype.h>

bool luhn(const char *num) {
    if (num == NULL) return false;

    size_t len = 0;
    while (num[len] != '\0') {
        len++;
    }

    if (len < 2) return false;

    int sum = 0;
    bool alt = false;

    for (size_t i = len; i-- > 0;) {
        if (num[i] == ' ') continue;

        if (!isdigit(num[i])) return false;

        int digit = num[i] - '0';

        if (alt) {
            digit *= 2;
            if (digit > 9) {
                digit -= 9;
            }
        }

        sum += digit;
        alt = !alt;
    }

    return (sum % 10 == 0);
}